android/src/main/kotlin/io/didomi/fluttersdk/EntitiesHelper.kt

```diff
@@ -0,0 +1,38 @@
+package io.didomi.fluttersdk
+import com.google.gson.Gson
+import io.didomi.sdk.Purpose
+import io.didomi.sdk.Vendor
+import java.util.*
+
+// Class used to convert Didomi entities (Vendor, Purpose) from and to types that can be sent through the Flutter channels.
+class EntitiesHelper {
+    companion object {
+        private val gson = Gson()
+        
+        // Convert a set of purposes into a list of maps.
+        @kotlin.jvm.JvmName("toListOfPurposes")
+        fun toList(purposes: Set<Purpose>): List<*> {
+            val json = gson.toJsonTree(purposes)
+            return gson.fromJson(json, List::class.java)
+        }
+
+        // Convert a set of vendors into a list of maps.
+        @kotlin.jvm.JvmName("toListOfVendors")
+        fun toList(vendors: Set<Vendor>): List<*> {
```

Comment:
Did you check that every vendors / purposes method return a non-null object? Maybe it is better to handle null lists just in case